alf/algorithms/agent.py

```diff
@@ -53,10 +50,10 @@ class Agent(OnPolicyAlgorithm):
     def __init__(self,
                  observation_spec,
                  action_spec,
+                 env=None,
+                 goal_generator=None,
                  rl_algorithm_cls=ActorCriticAlgorithm,
-                 encoding_network: Network = None,
                  intrinsic_curiosity_module=None,
```

Comment:
suggest to change a name for `intrinsic_curiosity_module`, as it suggests a specific algorithm. `intrinsic_reward_module` might be better.